### PR TITLE
add fetching filename in file search

### PR DESCRIPTION
**Pull request by pbolduc**
# Description

This PR includes the following proposed change(s):

- Returns the filename on file search.
- Filters out internal metadata from metadata returned
- Does not allow setting internal metadata via metadata dictionary

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/TrafficCourts/TrafficCourts.Coms.Client/ComsJsonContext.cs [hunk 1]
```diff
@@ -4,6 +4,8 @@ namespace TrafficCourts.Coms.Client;
 
 [JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase)]
 [JsonSerializable(typeof(ObjectMetadataCollection))]
+//[JsonSerializable(typeof(ObjectMetadata))]
+//[JsonSerializable(typeof(MetadataItem))]
```
Comment (pbolduc) [inline src/backend/TrafficCourts/TrafficCourts.Coms.Client/ComsJsonContext.cs:8]: Oops, I should remove these. Was troubleshooting the deserialization.

File: src/backend/TrafficCourts/TrafficCourts.Coms.Client/ObjectManagementClient.cs [hunk 2]
```diff
@@ -14,7 +14,8 @@ private ComsJsonContext JsonSerializerContext
     {
         get
         {
-            _jsonContext ??= new ComsJsonContext(JsonSerializerSettings);
+            var copy = new JsonSerializerOptions(JsonSerializerSettings);
```
Comment (pbolduc) [inline src/backend/TrafficCourts/TrafficCourts.Coms.Client/ObjectManagementClient.cs:17]: Should only create the copy in the ??= statement.

**Review by karahanb — APPROVED** [review]
Looks great but there are some build errors.

**Comment by pbolduc** [comment]
I will address it when I start working today. Odd I did not catch it running locally.

**Pull request merged**